GlotPress/GlotPress-WP
# Local: Add GP_SCRIPT_DEBUG to load the unminified CSS and JS

## What?

This PR improves the workflow for the unminified CSS and JS.

Resolves https://github.com/GlotPress/GlotPress/issues/1587.

## Why?


This PR resolves 2 problems:
* When a developer updates CSS or JS to a PR, she needs to add the minified files to the PR, so the reviewer can review the PR.
* When you clone the repository, you need to know that for loading the unminified assets (CSS and JS), you have to activate the SCRIPT_DEBUG constant in the wp-config.php file, so if you want to improve some CSS or JS, you have to activate this flag.


## How?

This PR proposes to add a new constant `GP_SCRIPT_DEBUG`, set to true by default, so:
* The new PRs don't need to commit the minified files.
* The users that clone the repository will use the unminified files by default.

When we will create a new release, we will need to add 2 steps to the release process:
* change this new constant `GP_SCRIPT_DEBUG` to false.
* MInify the assets with the Grunt task: `npm run build`.

This PR adds 2 new Grunt actions:
* the `browserify` action, to create the inline-translation.js file.
* the `copy` action, to copy the CSS and JavaScript files to the assets' folder (`assets/css` and `assets/js`).

## Testing Instructions





1. Execute `npm update` to add the new dependencies.
1. Execute the `npm run build` task to get the minified versions and to copy the JS and CSS files to the new destinations.
1. Set `SCRIPT_DEBUG` (wp-config.php file) and `GP_SCRIPT_DEBUG` (glotpress.php) to `false`, and check that you receive the minified files.
1. Set `SCRIPT_DEBUG` to `true` and `GP_SCRIPT_DEBUG` to `false`, and check that you receive the unminified files.
1. Set `SCRIPT_DEBUG` to `false` and `GP_SCRIPT_DEBUG` to `true`, and check that you receive the unminified files.
1. Set `SCRIPT_DEBUG` to `true` and `GP_SCRIPT_DEBUG` to `true`, and check that you receive the unminified files.

Please, check you have these files:

* Minified version (WP backend):
    * assets/css/inline-translation-loader.min.css
    * assets/css/inline-translation.min.css
    * assets/js/inline-translation-loader.min.js
    * assets/js/inline-translation.min.js
    
* Minified version (translation backend):
    * assets/css/style.min.css
    * assets/js/vendor/jquery.tablesorter.min.js
    * assets/js/common.min.js
    * assets/js/editor.min.js

* Unminified version (WP backend):
    * assets/css/inline-translation-loader.css
    * assets/css/inline-translation.css
    * assets/js/inline-translation-loader.js
    * inline-translation/inline-translation.js

* Unminified version (translation backend):
    * assets/css/style.css
    * assets/js/vendor/jquery.tablesorter.js
    * assets/js/common.js
    * assets/js/editor.js
